README.md

```diff
@@ -90,9 +65,61 @@ and launch training:
     ```bash
         export PROJECT_ID=<your project ID>
         export BUCKET_NAME=<your choice of a globally unique bucket ID>
-        gcloud alpha storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=us-central1 --uniform-bucket-level-access
+        gcloud storage buckets create gs://$BUCKET_NAME --project=$PROJECT_ID --location=us-central1 --uniform-bucket-level-access
     ```
 
+
+### Megatron-Deepspeed GPT Pretraining Instructions
+
+    
+The following instructions show how to train a GPT model using Megatron-Deepspeed 
+on a 96 H100 GPU cluster. The intruction are simplified to use Goolge Cloud Compute Engine APIs
+ only. In addition they use Google Cloud Strorage for storing the data:
+
+* **Task**: LLM Pretraining
+* **Implementation**: Megatron-Deepspped
+* **Distributed Framework**: Deepspeed (ZeRO stage 3)
+* **Infrastructure**: Google Cloud
+* **Cluster Management**: AI Infra cluster provisioning tool
+* **Storage**: Google Cloud Storage
+* **Dataset**: Wikipedia
+* **Model Architecture**: GPT 3
+* **Model Size**: 176B
+
+
+1.  **Download and Preprocess** Wikipedia dataset : 
+
+    ```bash
+        docker run gcr.io/llm-containers/gpt_preprocess:release ./preprocess.sh gs://$BUCKET_NAME
+    ```    
+
+    Warning: This could take hours to finish running.
+
+2.  **Pretrain GPT 176B on an A3 VM cluster**
+
+    ```bash
+        sudo docker run -it gcr.io/llm-containers/batch:release $PROJECT_ID gcr.io/llm-containers/gpt_train:release gs://$BUCKET_NAME 0 0 0 ' {"data_file_name":"wiki_data_text_document", "tensor_parallel":4, "pipeline_parallel":12, "nlayers":70, "hidden":14336, "heads":112, "seq_len":2048, "train_steps":100, "eval_steps":10, "micro_batch":1, "gradient_acc_steps":128 }' '{ "name_prefix": "megatron-gpt", "zone": "us-central1-a", "node_count": 12, "machine_type": "a3-highgpu-8g", "gpu_type": "nvidia-h100-80gb", "gpu_count": 8 }'
```

Comment:
Where did you get the 176B model parameters from (the nlayers, hidden, heads, seq_len)?

Reply:
This is similar to BLOOM https://huggingface.co/bigscience/bloom#model-architecture-and-objective

Reply:
Any particular reason you're going with that BLOOM model (which is modified from Megatron-LM GPT 2), rather than directly using the 175B GPT3 model config from Megatron Deepspeed (Page 10 of https://arxiv.org/pdf/2205.05198.pdf)?

Reply:
175B GPT3 model config won't fit in 12 nodes.  176B BLOOM model config would fit. 
BLOOM was trained using Megatron-deepseed. GPT3 was trained using Megatron-LM.  Both are based on the same GPT2 model architecture.